File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr) {
    const result = [];

    for (let i = 0; i < arr.length; i++) {
        // 이전 값과 다를 때만 추가
        if (result[result.length - 1] !== arr[i]) {
            result.push(arr[i]);
        }
    }

    return result;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        List<Integer> list = new ArrayList<>();
        
        int prev = -1;
        for (int num : arr){
            if (num != prev){
                list.add(num);      
            }
            prev = num;
        }
        
        return list.stream().mapToInt(i -> i).toArray();
    }
}